go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -25,6 +28,38 @@ const (
 	egressFirewallUpdateError      = "EgressFirewall Rules not correctly updated"
 )
 
+//egressfirewallDNSInfo stores information about the dnsNames used by egressfirewalls
+// this information applies to all egressfirewalls in all namespaces
+type egressfirewallDNSInformation struct {
+	sync.Mutex
+	//map of dnsNames to relevant information
+	dnsNameInfo map[string]*dnsInformation
+}
+
+func newEgressFirewallDNSInfo() egressfirewallDNSInformation {
+	return egressfirewallDNSInformation{
+		dnsNameInfo: make(map[string]*dnsInformation),
+	}
+}
+
+//dnsInformation stores information relating to a dns names
+type dnsInformation struct {
+	// the namespaces that have egressfirewall rules relating to this dnsName
+	// using a map of empty structs becuase the typical operation is search
+	Namespaces map[string]struct{}
```

Comment:
use sets.String

Reply:
(Looked to verify this was the same thing :) -> https://github.com/kubernetes/apimachinery/blob/v0.20.5/pkg/util/sets/string.go#L27